pkg/assisted/agentclusterinstall_test.go

```diff
@@ -0,0 +1,1180 @@
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithAdditionalAPIVip(testCase.ip)
+		assert.Equal(t, testCase.expectError, testBuilder.errorMsg)
+
+		if testCase.expectError == "" {
+			assert.Contains(t, testBuilder.Definition.Spec.APIVIPs, testCase.ip)
+		}
+	}
+}
+
+func TestWithIngressVip(t *testing.T) {
+	testCases := []struct {
+		ip          string
+		expectError string
+	}{
+		{
+			ip:          "192.168.1.10",
+			expectError: "",
+		},
+		{
+			ip:          "fd2e::10",
+			expectError: "",
+		},
+		{
+			ip:          "notanip",
+			expectError: "agentclusterinstall ingressVIP incorrectly formatted",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithIngressVip(testCase.ip)
+		assert.Equal(t, testCase.expectError, testBuilder.errorMsg)
+
+		if testCase.expectError == "" {
+			assert.Equal(t, testCase.ip, testBuilder.Definition.Spec.IngressVIP)
+		}
+	}
+}
+func TestWithAdditionalIngressVip(t *testing.T) {
+	testCases := []struct {
+		ip          string
+		expectError string
+	}{
+		{
+			ip:          "192.168.1.10",
+			expectError: "",
+		},
+		{
+			ip:          "fd2e::10",
+			expectError: "",
+		},
+		{
+			ip:          "notanip",
+			expectError: "agentclusterinstall ingressVIP incorrectly formatted",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithAdditionalIngressVip(testCase.ip)
+		assert.Equal(t, testCase.expectError, testBuilder.errorMsg)
+
+		if testCase.expectError == "" {
+			assert.Contains(t, testBuilder.Definition.Spec.IngressVIPs, testCase.ip)
+		}
+	}
+}
+func TestWithUserManagedNetworking(t *testing.T) {
+	testCases := []struct {
+		umn bool
+	}{
+		{
+			umn: true,
+		},
+		{
+			umn: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithUserManagedNetworking(testCase.umn)
+
+		assert.Equal(t, testCase.umn, *testBuilder.Definition.Spec.Networking.UserManagedNetworking)
+	}
+}
+func TestWithPlatformType(t *testing.T) {
+	testCases := []struct {
+		platform hiveextV1Beta1.PlatformType
+	}{
+		{
+			platform: "BareMetal",
+		},
+		{
+			platform: "None",
+		},
+		{
+			platform: "VSphere",
+		},
+		{
+			platform: "Nutanix",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := generateTestBuilder()
+		testBuilder.WithPlatformType(testCase.platform)
+
+		assert.Equal(t, testCase.platform, testBuilder.Definition.Spec.PlatformType)
+	}
+}
+func TestWithControlPlaneAgents(t *testing.T) {
+	testCases := []struct {
+		count         int
+		expectedError string
+	}{
+		{
+			count:         3,
+			expectedError: "",
+		},
+		{
+			count:         1,
+			expectedError: "",
+		},
+		{
+			count:         0,
+			expectedError: "agentclusterinstall controlplane agents must be greater than 0",
+		},
+		{
+			count:         -1,
```

Comment:
isn't this one redundant?

Reply:
0 and -1 are used to catch edged cases. If WithControlPlaneAgents has it's validation for agentCount removed or changed to allow 0, these tests will catch it.